File: lib/app/data/models/outfit.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'outfit.freezed.dart';
part 'outfit.g.dart';

enum OutfitType {
  casual,
  professional;

  String get displayName {
    switch (this) {
      case OutfitType.casual:
        return 'Casual';
      case OutfitType.professional:
        return 'Professional';
    }
  }
}

@freezed
class Outfit with _$Outfit {
  const factory Outfit({
    required String id,
    required String name,
    required OutfitType type,
    @Default('') String top,
    @Default('') String bottom,
    @Default('') String shoes,
    @Default('') String layer, // jacket/blazer
    @Default('') String accessories,
    @Default('') String notes,
    DateTime? lastWorn,
    @Default(0) int timesWorn,
  }) = _Outfit;

  factory Outfit.fromJson(Map<String, dynamic> json) => _$OutfitFromJson(json);
}

File: backend/app/models/outfit.py
"""Outfit model matching Flutter's outfit.dart."""

from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field


class OutfitType(str, Enum):
    """Outfit type."""
    casual = "casual"
    professional = "professional"


class OutfitBase(BaseModel):
    """Base outfit fields."""
    name: str = Field(..., min_length=1, max_length=100)
    type: OutfitType = OutfitType.casual
    top: str = ""
    bottom: str = ""
    shoes: str = ""
    layer: str = ""  # jacket/blazer
    accessories: str = ""
    notes: str = ""


class OutfitCreate(OutfitBase):
    """Outfit creation request."""
    pass


class OutfitUpdate(BaseModel):
    """Outfit update request."""
    name: str | None = Field(None, min_length=1, max_length=100)
    type: OutfitType | None = None
    top: str | None = None
    bottom: str | None = None
    shoes: str | None = None
    layer: str | None = None
    accessories: str | None = None
    notes: str | None = None


class Outfit(OutfitBase):
    """Outfit response model."""
    id: str
    user_id: str
    last_worn: datetime | None = None
    times_worn: int = 0
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        use_enum_values = True

